### ISSUE TITLE
[Feature Request]: Enable display of @Description annotations in generated Javadocs

### ISSUE THREAD

**Issue by slilichenko** [apache/beam#24265]
### What would you like to happen?

`*Option` classes contain `@Description` annotations which contain the details about the given option. But these annotations are no displayed in the generated Javadoc leading to poorly documented classes. 

### Issue Priority

Priority: 2

### Issue Component

Component: release

**Comment by slilichenko** [comment]
Yes, that's the idea - have a single source of truth and have it visible.

**Comment by lukecwik** [comment]
The textual representations used in `@Description` will limit what we can display to the user since there are places where we need to format it so that it displays correctly as javadoc. Take a look at a few of the differences in SdkHarnessOptions.java like https://github.com/apache/beam/blob/e6e832b62ec2b1f1ef5214fd4174d1362d444dce/sdks/java/core/src/main/java/org/apache/beam/sdk/options/SdkHarnessOptions.java#L91

**Comment by slilichenko** [comment]
Yes, I saw them. You also have cases like

[BigQueryOptions](https://github.com/apache/beam/blob/8df6f67c65b4888c45c31e088fb463972c4ec76b/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryOptions.java), which has no JavaDoc, but uses JavaDoc style in `@Description`. Perhaps that can be a middle ground, admittedly imperfect. Also, I am not sure that customers really use `./run-pipeline.sh --help=SomeSDKOptionClass` to get information about an option. It would be a Google or code search, at least IMHO.

**Comment by Abacn** [comment]
I tried to add "@Documented" to "@Description". The javadoc now contains the description, but the rendering is overflow for long description:

![image](https://github.com/apache/beam/assets/8010435/45850460-4692-42b9-b341-f91b0742b3b3)

Searched around and this seems not trivial
